test.js

```diff
@@ -16,212 +16,150 @@ const expect = Code.expect
 
 function getServer (port) {
   port = port || 4000
-  const server = new Hapi.Server()
-  server.connection({ port: port })
+  const server = new Hapi.Server({ port: port })
+
   return server
 }
 
-function start (server, opts, cb) {
-  if (typeof opts === 'function') {
-    cb = opts
-    opts = {}
-  }
+async function start (server, opts) {
+  opts = opts || {}
 
   const plugin = {
-    register: Multines.register,
+    plugin: {
+      name: 'multines',
+      register: Multines.register
+    },
     options: opts
   }
 
-  server.register([Nes, plugin], (err) => {
-    if (err) {
-      return cb(err)
-    }
-
-    server.start(cb)
-  })
+  await server.register([Nes, plugin])
 
   server.subscriptionFar('/echo')
   server.route({
     path: '/echo',
     method: 'POST',
-    handler: (req, reply) => {
+    handler: async (req, h) => {
       server.publishFar('/echo', req.payload)
-      reply(req.payload)
+
+      return req.payload
     }
   })
 
+  await server.start()
+
   return server
 }
 
 function pubSubTest () {
-  test('pub/sub', (done) => {
+  test('pub/sub', async () => {
     const client = new Nes.Client('ws://localhost:4000')
+    await client.connect()
 
-    client.connect((err) => {
-      if (err) {
-        return done(err)
-      }
-
-      client.subscribe('/echo', (message) => {
+    await new Promise((resolve, reject) => {
+      function handler (message, flags) {
         expect(message).to.equal({ hello: 'world' })
-        setTimeout(function () {
-          client.disconnect()
-          done()
-        }, 100)
-      }, (err) => {
-        if (err) {
-          return done(err)
-        }
 
-        client.request({
-          path: '/echo',
-          method: 'POST',
-          payload: { hello: 'world' }
-        }, (err) => {
-          if (err) {
-            return done(err)
-          }
+        client.disconnect().then(resolve)
+      }
+
+      return client.subscribe('/echo', handler)
+        .then(() => {
+          return client.request({
+            path: '/echo',
+            method: 'POST',
+            payload: { hello: 'world' }
+          })
```

Comment:
can you use await instead of then?

Reply:
I'm not happy with that either, forget about it. Lab@15 and async await are not great of testing this pattern.

Reply:
Side note, this does not seem correct either. It's *very hard* to test this pattern with async/await, and I would just want another testing library at this point.

Do:

```js
let resolve
let reject

  function handler (message, flags) {
    expect(message).to.equal({ hello: 'world' })

    client.disconnect().then(resolve).catch(reject)
  }

const p = new Promise((r1, r2) => {
  resolve = r1
  reject = r2
})

await client.subscribe('/echo', handler)
await Promise.all([client.request({ path: '/echo', ... }), p])
```

Reply:
@p16 can you do the above?